AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -188,18 +220,20 @@ internal class TestRunnerServiceImpl internal constructor(
     ): List<TestRunnerService.TestRunResult>? {
         if (!testMatrix.isComplete()) return null
         val resultPath = GcsPath(testMatrix.resultStorage.googleCloudStorage.gcsPath)
-        return findResultFiles(resultPath)
+        return findResultFiles(resultPath, testMatrix)
     }
 
     companion object {
         private const val MERGED_TEST_RESULT_SUFFIX = "-test_results_merged.xml"
         private const val LOGCAT_FILE_NAME = "logcat"
+        private const val LOGCAT_FILE_NAME_SUFFIX = "_logcat"
         private const val TEST_RESULT_XML_PREFIX = "test_result_"
         private const val TEST_RESULT_XML_SUFFIX = ".xml"
         private const val INSTRUMENTATION_RESULTS_FILE_NAME = "instrumentation.results"
     }
 
-    private class ResultFileResourceImpl(
+    @JsonClass(generateAdapter = true)
```

Comment:
i don't think we use generated adapters in this project, it is all reflection so you can remove this.